symtab.h

```diff
@@ -66,6 +66,10 @@ inline bool operator==(const Symbol& x, const Symbol& y) {
   return x.val() == y.val();
 }
 
+inline bool operator<(const Symbol& x, const Symbol& y) {
+  return x.str() < y.str();
```

Comment:
It seems this change makes this step slower (~5 secs to ~6 secs for me). Do we really want alphabetical ordering? I guess we just want build.ninja not to change significantly for small Makefile changes. I think x.val() < y.val() should suffice for this purpose. Even with x.val() < y.val(), there seems to be a very small performance regression (~0.1 sec maybe), but I think this is OK.